modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java

```diff
@@ -48,6 +48,9 @@
     /** Flag indicating if the cluster in read-only mode. */
     private final boolean readOnly;
 
+    /** Last cluster activation time. */
+    private final long activationTime;
```

Comment:
Looks like this is not activation time, but time when state changed for the last time. May be we should name it stateChangeTime or something like this.

Reply:
but it’s better to correct the condition on GridClusterStateProcessor.java:642. We should get time from message only if msg.activate() is true and state.active() is false, if I understood the idea correctly